test/unit/model/validation.test.js

```diff
@@ -266,10 +269,95 @@ describe(Support.getTestDialectTeaser('InstanceValidator'), function() {
   });
 
   describe('datatype validations', function () {
-    describe('should throw validationerror', function () {
-      var User = current.define('user', {
-        age: Sequelize.INTEGER
+    current = Support.createSequelizeInstance({
+      validation: true
+    });
+
+    var User = current.define('user', {
+      age: Sequelize.INTEGER,
+      name: Sequelize.STRING,
+      awesome: Sequelize.BOOLEAN,
+      number: Sequelize.DECIMAL,
+      uid: Sequelize.UUID
+    });
+
+    before(function () {
+      this.stub = sinon.stub(current, 'query', function () {
+        return new Promise(function (resolve) {
+          resolve(User.build({}));
+        });
       });
+    });
+
+    after(function () {
+      this.stub.restore();
+    });
+
+    describe('should not throw', function () {
+      describe('create', function () {
+        it('should allow number as a string', function () {
+          return expect(User.create({
+            age: '12'
+          })).not.to.be.rejected;
+        });
+
+        it('should allow decimal as a string', function () {
+          return expect(User.create({
+            number: '12.6'
+          })).not.to.be.rejected;
+        });
+
+        it('should allow string as a number', function () {
+          return expect(User.create({
+            name: 12
+          })).not.to.be.rejected;
+        });
+
+        it('should allow 0/1 as a boolean', function () {
+          return expect(User.create({
+            awesome: 1
+          })).not.to.be.rejected;
+        });
+
+        it('should allow 0/1 string as a boolean', function () {
+          return expect(User.create({
+            awesome: '1'
+          })).not.to.be.rejected;
+        });
+
+        it('should allow true/false string as a boolean', function () {
+          return expect(User.create({
+            awesome: 'true'
+          })).not.to.be.rejected;
+        });
```

Comment:
0/1 integers are ok, not strings. My personal opinion obviously but its just slightly too magic for me.
Although i honestly don't know how PG handles these cases (we might want to adjust for that).